### PR TITLE
libbpf-tools: Fixes for 32-bit platforms

### PR DESCRIPTION
**Pull request by jeez**
Fix build errors when compiling for 32-bit platforms by making the code more portable.

### CODE REVIEW

**Comment by jeez** [comment]
cc @anakryiko too since you've reviewed similar fixes in the past.

File: libbpf-tools/trace_helpers.c [hunk 1]
```diff
@@ -552,7 +553,8 @@ static int create_tmp_vdso_image(struct dso *dso)
 		return -1;
 
 	while (true) {
-		ret = fscanf(f, "%lx-%lx %*s %*x %*x:%*x %*u%[^\n]",
+		ret = fscanf(f,
+			     "%" PRIx64 "-%" PRIx64 "%*s %*x %*x:%*x %*u%[^\n]",
```
Comment (anakryiko) [inline libbpf-tools/trace_helpers.c:557]: I really dislike those PRIxxx macros, in libbpf we just explicitly cast to (long long) and use %lld, can you please do the same here?

File: libbpf-tools/trace_helpers.c [hunk 2]
```diff
@@ -669,10 +671,11 @@ struct syms *syms__load_file(const char *fname)
 		goto err_out;
 
 	while (true) {
-		ret = fscanf(f, "%lx-%lx %4s %lx %lx:%lx %lu%[^\n]",
-			     &map.start_addr, &map.end_addr, perm,
-			     &map.file_off, &map.dev_major,
-			     &map.dev_minor, &map.inode, buf);
+		ret = fscanf(f,
+			     "%" PRIx64 "-%" PRIx64 "%4s %" PRIx64 "%" PRIx64
+			     ":%" PRIx64 "%" PRIu64 "%[^\n]",
```
Comment (anakryiko) [inline libbpf-tools/trace_helpers.c:676]: ditto

**Review by anakryiko — CHANGES_REQUESTED** [review]

File: libbpf-tools/trace_helpers.c (hunk 1 above)
Comment (jeez) [reply libbpf-tools/trace_helpers.c:557]: I don't love them either but these are used in several other source files in this code base. Shouldn't we keep things consistent here?

Comment (anakryiko) [reply libbpf-tools/trace_helpers.c:557]: where do you see it used in libbpf-tools tools?

File: libbpf-tools/bindsnoop.c [hunk 3]
```diff
@@ -168,10 +168,10 @@ static void handle_event(void *ctx, int cpu, void *data, __u32 data_sz)
 		i++;
 	}
 	if (e->ver == 4) {
-		inet_ntop(AF_INET, &e->addr, addr, sizeof(addr));
-	} else {
-		inet_ntop(AF_INET6, &e->addr, addr, sizeof(addr));
-	}
+		inet_ntop(AF_INET, e->addr, addr, sizeof(addr));
+        } else {
```
Comment (chenhengqi) [inline libbpf-tools/bindsnoop.c:172]: Please use tab for indention.

File: libbpf-tools/trace_helpers.c (hunk 1 above)
Comment (jeez) [reply libbpf-tools/trace_helpers.c:557]: I meant in the root project, bcc.

File: libbpf-tools/bindsnoop.c (hunk 3 above)
Comment (jeez) [reply libbpf-tools/bindsnoop.c:172]: will fix in v2, thanks

File: libbpf-tools/trace_helpers.c (hunk 2 above)
Comment (jeez) [reply libbpf-tools/trace_helpers.c]: fixed in v2

**Review by anakryiko — APPROVED** [review]
LGTM, thanks.

**Pull request merged**